### PR TITLE
Fix #843 (--nospaceoperators doesn't work with closure RHS)

### LINKED ISSUE #843: --nospaceoperators doesn't work with closure RHS (and maybe LHS)
In my project I have a an infix `..` operator allowing me to configure any instance on creation:

```swift

let label = UILabel()..{
  $0.textColor = .red
}

// instead of

let label = {
  let label = UILabel()
  label.textColor = .red
  return label
}()
```

Until now, the `spaceAroundOperators` rule formatted it like that:
```swift
let label = UILabel() .. {
  $0.textColor = .red
}
```

But I just discovered the `--nospaceoperators` option, which I tried, but unfortunately it formats the code like that:
```swift
let label = UILabel().. {
  $0.textColor = .red
}
```
Hence, the operator is considered as a _postfix_ operator instead of an _infix_ operator, breaking my code.
I guess it conflicts with the `spaceAroundBraces` rule.

I didn't try with a LHS closure but if my assumption is correct, I guess there's the same issue.

### PR DESCRIPTION
**Pull request by jegnux**
This PR aims to fix #843 by adding a check that an opening brace isn't preceded by an infix operator. I also added a bunch of unit tests checking the `spaceAroundOperators` rule works even with weird lot or right hand sides (array literal, tuple, closure).

### CODE REVIEW

File: Sources/Rules.swift [hunk 1]
```diff
@@ -349,7 +349,7 @@ public struct _FormatRules {
         formatter.forEach(.startOfScope("{")) { i, _ in
             if let prevToken = formatter.token(at: i - 1) {
                 switch prevToken {
-                case .space, .linebreak,
+                case .space, .linebreak, .operator(_, .infix),
```
Comment (nicklockwood) [inline Sources/Rules.swift:352]: Why isn't this check also needed for the nextToken case?

Also, is there potentially also the same issue for custom postfix operators?

```swift
let foo = { bar }..
```

Comment (jegnux) [reply Sources/Rules.swift:352]: > Why isn't this check also needed for the nextToken case?
That's a good question, I was as surprised as you are, but unfortunately I don't have the answer :-/ 

> Also, is there potentially also the same issue for custom postfix operators?
Good point, I will add a bunch of test to check that as well 👍

**Comment by jegnux** [comment]
@nicklockwood Updated :)

**Comment by nicklockwood** [comment]
Looks great - thanks 👍

**Pull request merged**